lib/wallet/src/Cardano/Wallet/Primitive/AddressDerivation/Icarus.hs

```diff
@@ -446,3 +447,6 @@ instance PersistPublicKey (IcarusKey depth) where
       where
         xpubFromText = CC.xpub <=< fromHex @ByteString
         err _ = error "unsafeDeserializeXPub: unable to deserialize IcarusKey"
+
+instance NetworkDiscriminantCheck (n :: NetworkDiscriminant) IcarusKey where
+    networkDiscriminantCheck _ = True
```

Comment:
Maybe worth a comment explaining why we don't validate in the case of Icarus keys?

Reply:
done